src/main/java/subway/line/LineQuery.java

```diff
@@ -0,0 +1,28 @@
+package subway.line;
+
+import org.springframework.stereotype.Component;
+import subway.exception.LineNotFoundException;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class LineQuery {
+    private LineRepository lineRepository;
+
+    public LineQuery(LineRepository lineRepository) {
+        this.lineRepository = lineRepository;
+    }
+
+    public List<Line> findAll() {
+        return lineRepository.findAll();
+    }
+
+    public Line findById(long lineId) {
+        Optional<Line> OptionalLine = lineRepository.findById(lineId);
+        if (OptionalLine.isEmpty()) {
+            throw new LineNotFoundException(lineId);
+        }
+        return OptionalLine.get();
+    }
```

Comment:
다음과 같이 개선해 볼 수 있을 것 같아요 😄 
```suggestion
    public Line findById(long lineId) {
        public Line findById(long lineId) {
        return lineRepository.findById(lineId)
                .orElseThrow(() -> new LineNotFoundException(lineId));
    }
```